server/app/services/applications/ProgramAdminApplicationService.java

```diff
@@ -95,9 +99,13 @@ public void setStatus(
     // The send/sent phrasing is a little weird as the service layer is converting between intent
     // and reality.
     boolean sendEmail = newStatusEvent.emailSent();
+    ProgramDefinition programDef = programRepository.getShallowProgramDefinition(program);
```

Comment:
**suggestion (non-blocking):** In this case, I think we're better off making the call to [get the full program definition](https://github.com/civiform/civiform/blob/b75bc4e6dd27df9981036c3acc6e6d974b6eb46a/server/app/services/program/ProgramService.java#L175), since that will utilize the cache if its available. I imagine when we change this to allow for bulk status updates, calling to get the program definition on every status update will slow things down a lot. I know this was already being called each time, so this is somewhat outside the scope of this PR, but something to be aware of.

Reply:
This code is just a temporary hook Danielle. I plan to move this to its ApplicationStatusService class which will be called from the controller. Since this is getting the shallowProgramDef only for checking error state, does it matter?/

Reply:
It looks like its part of the `setStatus` function, which I assume gets called each time someone wants to update a status (meaning a database call for each application that the status is changing for). This can be part of a larger refactor I'll plan to do though, so no worries on changing it in this PR.